anta/cli/exec/utils.py

```diff
@@ -113,3 +120,78 @@ async def collect_commands(inv: AntaInventory,  enable_pass: str, commands: Dict
                 f"Exception raised for device {device.name} - {type(e).__name__}: {str(e)}"
             )
             logger.debug(traceback.format_exc())
+
+
+def device_directories_show_tech_support(dev: str, root_dir: str) -> Tuple[str, str]:
+    """
+    return a tuple containing the show_tech_directory and the device_directory
+    """
+    cwd = os.getcwd()
+    show_tech_directory = f"{cwd}/{root_dir}"
+    device_directory = f"{show_tech_directory}/{dev}"
+    for directory in [show_tech_directory, device_directory]:
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+    return show_tech_directory, device_directory
+
+
+def create_ssh_client(
+    dev: str, port: int, username: str, password: str, banner_timeout: int = 60
+) -> paramiko.SSHClient:
+    """
+    return a connected ssh client
+    """
+    client = paramiko.SSHClient()
+    client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    client.connect(
+        hostname=dev,
+        port=port,
+        username=username,
+        password=password,
+        banner_timeout=banner_timeout
+    )
+    return client
+
+
+async def collect_scheduled_show_tech(inv: AntaInventory, root_dir: str, tags: List[str], ssh_port: int = 22) -> None:
+    """
+    Collect scheduled show-tech on devices
+    """
+    logger.info("Connecting to devices...")
+    await inv.connect_inventory()
+    devices = inv.get_inventory(established_only=True, tags=tags)
+    if len(devices) > 0:
+        # Collect all the tech-support files stored on Arista switches flash and copy them locally
+        for device in devices:  # TODO: should use asyncio.gather instead of a loop.
+            try:
+                # Create one zip file named all_files.zip on the device with the all the show tech-support files in it
+                await device.session.cli(
+                    command=f"bash timeout 30 zip {ZIP_FILE} /mnt/flash/schedule/tech-support/*"
+                )
+                logger.info(f"Created {ZIP_FILE} on device {device.name}")
+                # Create directories
+                output_dir = device_directories_show_tech_support(
+                    device.name, root_dir)
+                # Connect to the dpreevice using SSH
+                ssh = create_ssh_client(
+                    device.host, ssh_port, device.username, device.password
+                )
+                # Get the zipped file all_files.zip using SCP and save it locally
+                my_path = f"{output_dir[1]}/{CURRENT_DATE}_{device.name}.zip"
+                scp = SCPClient(ssh.get_transport())
+                scp.get(ZIP_FILE, local_path=my_path)
+                scp.close()
```

Comment:
on doit pouvoir utiliser un context ici comme la classe implémente __enter__ et __exit__
https://github.com/jbardin/scp.py/blob/master/scp.py

```suggestion
                with SCPClient(ssh.get_transport()) as scp:
                    scp.get(ZIP_FILE, local_path=my_path)
```